stock_move_auto_assign/models/stock_move.py

```diff
@@ -13,10 +14,29 @@ class StockMove(models.Model):
 
     def _action_done(self, cancel_backorder=False):
         done_moves = super()._action_done(cancel_backorder=cancel_backorder)
-        done_moves._prepare_auto_assign()
+        done_moves._prepare_auto_assign(location_field="move_line_ids.location_dest_id")
         return done_moves
 
-    def _prepare_auto_assign(self):
+    def _action_cancel(self):
+        moves_with_reservation_to_cancel = self.filtered(
+            lambda m: m.state not in ("done", "cancel")
+            and float_compare(
+                m.reserved_availability, 0, precision_rounding=m.product_uom.rounding
+            )
+            > 0
+        )
+        res = super()._action_cancel()
+        moves_with_reservation_cancelled = (
```

Comment:
we should favor US english
```suggestion
        moves_with_reservation_canceled = (
```
same below

Reply:
done